### PR TITLE
Rest of Green Brin Runway migration

### PR DESCRIPTION
**Pull request by blkerby**
There is opportunity to add more leave charged/sparking strats, e.g. in Green Hill Zone for the top doors (and a ton of 3-room strats in Green Brinstar Main Shaft). This PR is mostly just trying to migrate the existing ones

### CODE REVIEW

File: region/brinstar/green/Noob Bridge aka A Bridge Too Far.json [hunk 1]
```diff
@@ -195,11 +120,41 @@
...
+      "requires": [
+        {"canShineCharge": {
+          "usedTiles": 33,
+          "openEnd": 2
+        }},
+        "canShinechargeMovement"
+      ],
+      "exitCondition": {
+        "leaveShinecharged": {
+          "framesRemaining": 120
```
Comment (osse101) [inline region/brinstar/green/Noob Bridge aka A Bridge Too Far.json:154]: At full run I was seeing ~140, but when shortcharging I could get 150.
So there is a full run vs shortcharge difference, and that would be something to cover with a `h_shinechargeLeniency` in the future.

File: region/brinstar/green/Spore Spawn Kihunter Room.json [hunk 2]
```diff
@@ -166,11 +120,41 @@
...
+        {"id": 2}
       ]
     }
   ],
   "strats": [
+    {
+      "link": [1, 1],
+      "name": "Leave With Runway",
+      "notable": false,
+      "requires": [],
```
Comment (osse101) [inline region/brinstar/green/Spore Spawn Kihunter Room.json:133]: This requires killing the left kihunter so it should have some basic requirement.

Comment (blkerby) [reply region/brinstar/green/Spore Spawn Kihunter Room.json]: Done.

File: region/brinstar/green/Noob Bridge aka A Bridge Too Far.json (hunk 1 above)
Comment (blkerby) [reply region/brinstar/green/Noob Bridge aka A Bridge Too Far.json]: There are a lot of other factors too that could justify shinecharge leniency:
- tendency for less experienced players to idle for a moment after getting the shinecharge, before beginning moving
- lack of knowledge/ability of how to ledge-grab, down-back, or other movement techniques
- lack of knowledge/ability to quickly hone in accurately on the place where the spark should be activated, e.g. by stopping on a dime or carefully controlling a jump.

I think we could treat this kind of like heat frames, where we put tight values into the strats, and based on settings, the randomizer assumes you need extra frames available for leniency, e.g. maybe something like this:

Basic: 60 frames
Medium: 30 frames
Hard: 20 frames
Very Hard: 10 frames
Expert: 5 frames
Extreme: 2 frames
Insane: 0 frames

In this case, the leniency wouldn't need to be explicitly written into the strats.

**Review by osse101 — APPROVED** [review]

**Pull request merged**